arch/arm/src/lpc17xx_40xx/lpc17_40_i2c.c

```diff
@@ -242,7 +244,18 @@ static int lpc17_40_i2c_start(struct lpc17_40_i2cdev_s *priv)
            lpc17_40_i2c_timeout, (wdparm_t)priv);
   nxsem_wait(&priv->wait);
 
-  return priv->nmsg;
```

Comment:
why not simplify to:
```
return priv->nmsg ? -ENXIO : 0;
```

Reply:
I thought that returning the number o messages transmitted was required, but digging a little bit in the source code revealed that it is optional according to the I2C master driver documentation:
https://github.com/apache/incubator-nuttx/blob/master/include/nuttx/i2c/i2c_master.h#L146-L154

In this case, your suggestion seems cleaner. I'll update the PR.